src/doc/src/reference/unstable.md

```diff
@@ -199,6 +199,36 @@ cargo +nightly build -Z config-profile
 ```
 
 
+### Build Profile
+* Tracking Issue: [rust-lang/rust#48683](https://github.com/rust-lang/rust/issues/48683)
+
+The `build` profile controls the build settings for build scripts,
+proc-macros, compiler plugins, and all of their dependencies. It requires the
+`build-profile` feature to be enabled.
+
+```toml
+cargo-features = ["build-profile"]
+
+[profile.build]
+# The following illustrates the defaults.
+opt-level = 0
+debug = false
+rpath = false
+lto = false
+debug-assertions = false
```

Comment:
I'm a little wary about this being `false`, it seems like this may want to be `true` by default to help weed out mistakes more quickly

Reply:
Yea, I wouldn't mind making it `true`. Maybe the same for `overflow-checks`?  I don't have a sense of how much slower that typically makes things, but I suspect it would not be perceivable by most scripts/macros.  I think `debug` is the bigger question of how it should be defaulted.

Reply:
Yeah I think this and overflow-checks should probably default to on, build scripts are typically rarely a bottleneck and if they both of these options can be disabled pretty easily (both from crates.io via changing apis or locally by changing profiles).

For `debug` I wonder if we could perhaps try setting a default of 1? That means we only generate line tables for backtraces, but no local variable info as no one's really using gdb on these